src/main/java/com/softteam/vocabuilder/controller/VocabularyController.java

```diff
@@ -0,0 +1,53 @@
+package com.softteam.vocabuilder.controller;
+
+import com.softteam.vocabuilder.persistence.entity.Vocabulary;
+import com.softteam.vocabuilder.persistence.repository.VocabularyRepository;
+import com.softteam.vocabuilder.service.VocabularyServiceImp;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/vocabulary")
+public class VocabularyController {
+
+    @Autowired
+    private VocabularyServiceImp vocabularyServiceImp;
+    @Autowired
+    private VocabularyRepository vocabularyRepository;
+
+    @PostMapping
+    public ResponseEntity<Vocabulary> createVocabulary(@RequestBody Vocabulary vocabulary){
+        vocabularyServiceImp.create(vocabulary);
+        return new ResponseEntity<Vocabulary>(HttpStatus.CREATED);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<Vocabulary> updateVocabulary(@PathVariable(value = "id")UUID id, @RequestBody Vocabulary vocabulary){
```

Comment:
I changed that, because I call the validate method UUID Type(String id) and pass String . This returns the UUID if it is valid or throws an error if it is a String that does not comply with the UUID format.